03-viz.Rmd

```diff
@@ -1,40 +1,528 @@
+
+```{r 03-data-co2-line-2}
+co2_line <- ggplot(co2_df, aes(x = date, y = concentration)) + 
+                   geom_line() +
+                   xlab('Year') +
+                   ylab('Atmospheric CO2 (ppm)') + 
+                   theme(text = element_text(size = 18))
+```
+
+```{r 03-data-co2-line-2-print, echo = FALSE, results = 'hide', message = FALSE, warning = FALSE}
+print(co2_line)
+```
+
+Finally, let's see if we can better understand the oscillation by changing the visualization a little bit. Note that it is 
+totally fine to use a small number of visualizations to answer different aspects of the question you are trying to answer. We will accomplish
+this by using *scales*, another important feature of `ggplot2` that allow us to easily transform the different variables and set limits.
+We scale the horizontal axis by using the `scale_x_continuous` function, and the vertical axis with the `scale_y_continuous` function.
+We can transform the axis by passing the `trans` argument, and set limits by passing the `limits` argument. In particular, here we 
+will use the `scale_x_continuous` function with the `limits` argument to zoom in on just five years of data (say, 1990-1995):
+
+```{r 03-data-co2-line-3}
+co2_line <- ggplot(co2_df, aes(x = date, y = concentration)) + 
+                   geom_line() +
+                   xlab('Year') +
+                   ylab('Atmospheric CO2 (ppm)') + 
+                   scale_x_continuous(limits = c(1990, 1995)) +
+                   theme(text = element_text(size = 18))
+```
+
+```{r 03-data-co2-line-3-print, echo = FALSE, results = 'hide', message = FALSE, warning = FALSE}
+print(co2_line)
+```
+
+Interesting! It seems that each year, the atmospheric CO2 increases until it reaches its peak somewhere around April, decreases until around late September, 
+and finally increases again until the end of the year. In Hawaii, there are two seasons: summer from May through October, and winter from November through April.
+Therefore, the oscillating pattern in CO2 matches up fairly closely with the two seasons.
+
+
+### The island landmass dataset
+This dataset contains a list of Earth's land masses as well as their area (in thousands of square miles). **Question:** Are the continents (North / South America, Africa, Europe, Asia, Australia, Antarctica) Earth's 7 largest landmasses? If so, what are the next few largest landmasses after those?
+```{r 03-data-islands}
+# islands data 
+# (with some light wrangling to make it a dataframe with informative column names)
+islands_df <- enframe(islands)
+colnames(islands_df) <- c('landmass', 'size')
+head(islands_df)
+```
+Here, we have a list of Earth's landmasses, and are trying to compare their sizes. The right type of visualization to answer this
+question is a bar plot, specified by the `geom_bar` geometric object in `ggplot2`. However, by default, `geom_bar` sets the heights
```

Comment:
```suggestion
question is a bar plot, specified by the `geom_bar` function in `ggplot2`. However, by default, `geom_bar` sets the heights
```
I think `geom_bar` is a function? It might return a `geom_bar` object (docs not super clear to me), but here I think it makes sense to refer to it as a function.

Reply:
changed this and made a similar change throughout to other instances